elasticsearch/query.go

```diff
@@ -123,9 +123,15 @@ func (ess *ElasticsearchStorage) QueryEvents(ctx context.Context, filter nostr.F
 	// optimization: get by id
 	if isGetByID(filter) {
 		if evts, err := ess.getByID(filter); err == nil {
-			for _, evt := range evts {
-				ch <- evt
-			}
+			go func() {
+				for _, evt := range evts {
+					select {
+					case ch <- evt:
+					case <-ctx.Done():
+						return
+					}
+				}
+			}()
 			close(ch)
```

Comment:
This seems to be occur panic with sending to closed channel.

Reply:
can you move this close into goroutine with defer close(ch) ?

Reply:
Moved.